### PR TITLE
Update ActiveStorage Attachment model documentation

### LINKED ISSUE #34298: Updated needed for ActiveStorage Attachment model documentation
I noticed that the documentation for the ActiveStorage `Attachment` model is not updated to reflect some recent changes. 

There is now a `ActiveRecord::InvalidForeignKey` check/rescue that prevents the blob from being destroyed if other attachments exist here. 

Current documentation:

> Attachments associate records with blobs. Usually that's a one record-many blobs relationship, but it is possible to associate many different records with the same blob. If you're doing that, you'll want to declare with <tt>has_one/many_attached :thingy, dependent: false</tt>, so that destroying any one record won't destroy the blob as well. (Then you'll need to do your own garbage collecting, though).

https://github.com/rails/rails/blob/4725b2f0bac7fd062ac03c19f1648011452363d2/activestorage/app/models/active_storage/attachment.rb#L5

PR that Implements the functionality update:
https://github.com/rails/rails/pull/33438

Line that implements the change:
https://github.com/rails/rails/blob/4725b2f0bac7fd062ac03c19f1648011452363d2/activestorage/app/models/active_storage/blob.rb#L36

I'm happy to submit a PR to update the docs, but I just wanted to run the updated text by someone first. I'm thinking it could be as simple as omitting the last few sentences, but here's a stab at adding some info to let people explicity know what's going on:

> Attachments associate records with blobs. Usually that's a one record-many blobs relationship, but it is possible to associate many different records with the same blob. In the event multiple attachments share the same blob, an <tt>ActiveRecord::ForeignKey</tt> check is performed so that destroy an attachment does not destroy the blob if it is being used by other attachments.

### PR DESCRIPTION
**Pull request by kylekeesling**
This PR updates some documentation to reflect changes made in this PR:
https://github.com/rails/rails/pull/33438

More details about this PR can be found here:
https://github.com/rails/rails/issues/34298

closes #34298

### CODE REVIEW

**Review by francescobbo — APPROVED** [review]

File: activestorage/app/models/active_storage/attachment.rb [hunk 1]
```diff
@@ -3,9 +3,9 @@
 require "active_support/core_ext/module/delegation"
 
 # Attachments associate records with blobs. Usually that's a one record-many blobs relationship,
-# but it is possible to associate many different records with the same blob. If you're doing that,
-# you'll want to declare with <tt>has_one/many_attached :thingy, dependent: false</tt>, so that destroying
-# any one record won't destroy the blob as well. (Then you'll need to do your own garbage collecting, though).
+# but it is possible to associate many different records with the same blob. A foreign-key constraint
+# on the attachments table prevents blobs from being purged if they’re still attached to any records.
+
```
Comment (robin850) [inline activestorage/app/models/active_storage/attachment.rb]: Could you remove this blank line please ?

Comment (kylekeesling) [reply activestorage/app/models/active_storage/attachment.rb]: @robin850 I just submitted an update removing the extra blank line

**Comment by robin850** [comment]
Thanks @kylekeesling ! In the future, you can add `[ci skip]` to your commit message to prevent Travis CI from running a build as this is not required for documentation changes. :-)

**Comment by kylekeesling** [comment]
@robin850 ok cool - thanks for the tip!

**Pull request merged**